impl/src/main/java/com/okta/sdk/impl/resource/AbstractPropertyRetriever.java

```diff
@@ -279,6 +279,17 @@ protected Map getMapProperty(String key) {
 
         if (value != null) {
             if (value instanceof String) {
+                if(type.getName().equals("com.okta.sdk.resource.application.ApplicationSignOnMode")) {
+                    E convertedValue;
+                    try {
+                        convertedValue = enumConverter.fromValue(type, value.toString());
+                    } catch (IllegalArgumentException e) {
+                        convertedValue = enumConverter.fromValue(type, "CUSTOM");
+                        String msg = "Undeclared enum value. Defaulting to CUSTOM";
+                        log.error(msg, e);
+                    }
+                    return convertedValue;
+                }
```

Comment:
This block is going to need comments to explain what is going on here.
Especially with a note to remove it when: #184 is completed
use `ApplicationSignOnMode.class` instead of the string name

Reply:
@bdemers We can't add this new app type to the spec, as its not documented. We are going to set it so a app type `signonmode` that doesn't exist in the mapping returns a `DefaultApplication` as we do in other SDK's